app/routes/blog/index.tsx

```diff
@@ -23,12 +26,15 @@ export const loader = async ({ request }: { request: Request }) => {
 export default function Blog() {
   const { page, posts, nextPage, previousPage, totalPages } =
     useLoaderData<LoaderData>();
+  const [query, setQuery] = useState("");
+  const controlledPosts = useSearchQuery(posts, query);
```

Comment:
This is only filtering the posts that are already displayed in the screen. We should filter using all the posts available. 

Try duplicating the posts in the app to reproduce this.

Reply:
Oh, just figured you must have filtered the posts already so as to display them in chunks.
Sorry, I didn't think of that earlier.